designer/server/src/main/scala/pl/touk/nussknacker/ui/additionalconfig/AdditionalUIConfigProvider.scala

```diff
@@ -0,0 +1,73 @@
+package pl.touk.nussknacker.ui.additionalconfig
+
+import pl.touk.nussknacker.engine.api.component.{
+  ComponentGroupName,
+  ComponentId,
+  ParameterConfig,
+  ScenarioPropertyConfig,
+  SingleComponentConfig
+}
+import pl.touk.nussknacker.engine.api.definition.FixedExpressionValue
+import pl.touk.nussknacker.engine.definition.component.parameter.editor.EditorExtractor
+import pl.touk.nussknacker.engine.graph.node.FragmentInputDefinition.{
+  ParameterValueCompileTimeValidation,
+  ValueInputWithFixedValues
+}
+import pl.touk.nussknacker.engine.util.Implicits.RichScalaMap
+import pl.touk.nussknacker.ui.additionalconfig.AdditionalUIConfigProvider.AdditionalUIConfig
+
+/**
+ * Trait allowing the provision of UI configuration for components and scenario properties, without requiring a model reload.
+ *
+ * TODO: The current implementation allows providing configs only for standard components - meaning that base components aren't handled.
+ */
+trait AdditionalUIConfigProvider extends Serializable {
+
+  def getAllForProcessingType(processingType: String): Map[ComponentId, AdditionalUIConfig]
+
+  def getScenarioPropertiesUIConfigs(processingType: String): Map[String, ScenarioPropertyConfig]
+
+}
+
+object AdditionalUIConfigProvider {
+  val empty = new DefaultAdditionalUIConfigProvider(Map.empty, Map.empty)
+
+  case class AdditionalUIConfig(
```

Comment:
Let's move these two classes outside the object. We used to do this way and after all, it was very messy, We very often had long versions like `AdditionalUIConfigProvider.AdditionalUIConfig` in imports, it was impossible to extract the class to a separate file without changing imports.